src/profile/ProfileForm.tsx

```diff
@@ -37,585 +23,315 @@ interface ChangedUser {
-                    errors: { status_code: 0, password: [] },
-                  });
-                  setShowPassword(true);
-                }}
-              >
-                Generate
-              </Button>
-            </Col>
-            <Col>
-              <InputGroup>
-                <Form.Control
-                  type={showPassword ? "text" : "password"}
-                  name="password"
-                  onChange={onChange}
-                  isInvalid={
-                    state.errors.password && state.errors.password.length > 0
-                  }
-                  value={state.changedUser?.password ?? ""}
-                />
-                <InputGroup.Append>
-                  <Button
-                    variant={showPassword ? "outline-primary" : "outline-secondary"}
-                    onClick={() => {
-                      setShowPassword((s) => !s);
-                    }}
-                  >
-                    <FontAwesomeIcon icon={showPassword ? faEyeSlash : faEye} />
-                  </Button>
-                </InputGroup.Append>
-              </InputGroup>
-              <Form.Control
-                type="hidden"
-                isInvalid={
-                  state.errors.password && state.errors.password.length > 0
+        <Col sm={2}>
+          <Button
+            className="w-100"
+            onClick={() => {
+              props.context.setValue("password", nanoid());
+              // set the password visible so you can see what you get
+              setShowPassword(true);
+            }}
+          >
+            Generate
+          </Button>
+        </Col>
+        <Col sm={7}>
+          <Field
+            errors={props.context.errors}
+            type={showPassword ? "text" : "password"}
+            name="password"
+            ref={props.context.register({
+              minLength: {
+                value: 8,
+                message: "USER.PASSWORD.TOO_SHORT.8",
+              },
+              validate: (password?: string) => {
+                // the pattern argument doesn't let us return an error if the value FAILS a regex,
+                // so we'll do it ourselves
+                if (password && /^\d*$/.test(password)) {
+                  return "USER.PASSWORD.ENTIRELY_NUMERIC";
                 }
-                value={state.changedUser?.password}
-              />
-              <Form.Control.Feedback type="invalid">
-                <ul>
-                  {state.errors.password?.map((msg) => (
-                    <li key={msg}>{ERRORS[msg]}</li>
-                  ))}
-                </ul>
-              </Form.Control.Feedback>
-            </Col>
-          </Form.Row>
+                if (props.context.getValues().cur_password && !password) {
+                  return "USER.PASSWORD.NEW_REQUIRED";
+                }
+              },
+            })}
+          />
+        </Col>
+        <Col sm={1}>
+          <Button
```

Comment:
The button should have clearly distinguishable on/off styles, and should be a secondary/tertiary button instead of a primary one.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/c57717f2d58a502ca8753aae95a11bdd6dbbe021